### PR TITLE
feat: Shadow tracking

### PR DESCRIPTION
**Pull request by staffik**
Part of: https://github.com/near/near-one-project-tracking/issues/65
An option for non-validator node to track shards of given validator.

During stateful -> stateless protocol upgrade a node will track all shards and will require a lot of RAM. After the migration we can move the validator key to a new, smaller node, that does not track all shards.
To make it with minimal downtime, the new node needs to have appropriate shards in place and memtries loaded in memory, then we hot swap the validator key without stopping the new node.
But before that happen the new node is not a validator and we need a way to tell it which validator's shards it should track.

### CODE REVIEW

File: core/chain-configs/src/client_config.rs [hunk 1]
```diff
@@ -410,6 +410,8 @@ pub struct ClientConfig {
     pub gc: GCConfig,
     /// Accounts that this client tracks.
     pub tracked_accounts: Vec<AccountId>,
+    /// Track shards that should be tracked by given validator.
+    pub shadow_tracked: Option<AccountId>,
```
Comment (wacban) [inline core/chain-configs/src/client_config.rs:414]: nit: How about tracked_shadow_validator? All the other relevant fields are prefixed tracked so I would stick with that and shadow alone doesn't describe well enough what this is about.

**Review by wacban — COMMENTED** [review]
Can you test that it works well when the shard assignment is changing and the node should state sync?

**Comment by staffik** [comment]
> Can you test that it works well when the shard assignment is changing and the node should state sync?

Tested in forknet20, I will add a dedicated pytest.

File: chain/epoch-manager/src/shard_tracker.rs [hunk 2]
```diff
@@ -90,6 +96,9 @@ impl ShardTracker {
                 let subset = &schedule[index as usize];
                 Ok(subset.contains(&shard_id))
             }
+            TrackedConfig::ShadowValidator(account_id) => {
+                self.epoch_manager.cares_about_shard_in_epoch(*epoch_id, account_id, shard_id)
```
Comment (tayfunelmas) [inline chain/epoch-manager/src/shard_tracker.rs:100]: can we also give some warning somewhere (potentially here) if the accountid is not eligible to be a chunk validator as all? I was thinking about when loading the config but at that point I am not sure if the node will have the full info about the other validators.

**Review by tayfunelmas — APPROVED** [review]

File: chain/epoch-manager/src/shard_tracker.rs (hunk 2 above)
Comment (staffik) [reply chain/epoch-manager/src/shard_tracker.rs:100]: Will look into it in a separate PR, together with nayduck test.

**Pull request merged**